rsa.go

```diff
@@ -397,15 +397,21 @@ func newRSAKey3(isPriv bool, N, E, D, P, Q, Dp, Dq, Qinv BigInt) (C.GO_EVP_PKEY_
 	}
 	comps = append(comps, required[:]...)
 
-	// OpenSSL 3.0 and 3.1 required all the precomputed values if
-	// P and Q are present. See:
-	// https://github.com/openssl/openssl/pull/22334
-	if P != nil && Q != nil && Dp != nil && Dq != nil && Qinv != nil {
-		precomputed := [...]bigIntParam{
-			{paramRSA_P, P}, {paramRSA_Q, Q},
-			{paramRSA_Dp, Dp}, {paramRSA_Dq, Dq}, {paramRSA_Qinv, Qinv},
+	if P != nil && Q != nil {
+		allPrecomputedExists := Dp != nil && Dq != nil && Qinv != nil
+		// OpenSSL 3.0 and 3.1 required all the precomputed values if
+		// P and Q are present. If they are not, we need to omit also P and Q.
+		// See https://github.com/openssl/openssl/pull/22334
```

Comment:
I had a hard time with the past tense "required" vs. present "are", and "they", and what OpenSSL versions are being discussed. This is how I interpret it:

```suggestion
		// The precomputed values should only be passed if P and Q are present
		// and every precomputed value is present. (If any precomputed value is
		// missing, don't pass any of them.)
		//
		// In OpenSSL 3.0 and 3.1, we must also omit P and Q if any precomputed
		// value is missing. See https://github.com/openssl/openssl/pull/22334
```